lib/rubocop/cop/performance/case_when_splat.rb

```diff
@@ -0,0 +1,77 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Performance
+      # Place `when` conditions that use `splat` at the end
+      # of the `when` branches.
+      #
+      # Ruby has to allocate memory for the splat expansion every time
+      # that the `case` `when` statement is run. Since Ruby does not support
+      # fall through inside of `case` `when`, like some other languages do,
+      # the order of the `when` branches does not matter. By ordering any
+      # splat expansions at the end of the list of `when` branches we will
+      # reduce the number of times that memory has to be allocated for
+      # the expansion.
+      #
+      # @example
+      #   # bad
+      #   case foo
+      #   when *[1, 2, 3, 4]
+      #     bar
+      #   when 3
+      #     baz
+      #   end
+      #
+      #   # good
+      #   case foo
+      #   when 3
+      #     baz
+      #   when *[1, 2, 3, 4]
```

Comment:
It's strange to use an example where rearranging the `when`s has changed the semantics. So remove `3` from the array.